### ISSUE TITLE
Images with github.com src not replaced

### ISSUE THREAD

**Issue by kasbah** [npm/marky-markdown#229]
GitHub will replace images with src of `github.com/<whatever>/<project>/blob/master/example.png` with a src of `raw.githubusercontent.com/<whatever>/<project>/master/example.png` but marky-markdown doesn't do that. 

Here is an example readme: https://github.com/MickMad/HACK/blob/96e62a68499308fb094ed7ab7ab57de1d313a980/README.md

**Comment by ashleygwilliams** [comment]
thanks for filing! we strive for github parity, so we consider this a bug- if you're interested in contributing, patches welcome!

**Comment by revin** [comment]
Hi @kasbah, thanks for filing!

Here's what I see on GitHub for the readme you linked:

<img width="1260" alt="screen shot 2016-08-12 at 5 46 43 pm" src="https://cloud.githubusercontent.com/assets/94850/17640468/ddd728ca-60b4-11e6-97f2-87e54bf19c9c.png">

It doesn't look like GitHub is modifying the link as you're expecting. I think if you update the image path such that it's relative to the README, GitHub will behave as you expect:

```markdown
![HACK top](graphics/hack01.png)

![HACK bottom](graphics/hack02.png)
```

Happily, marky does indeed support this use case, if you supply the package details in the `options`, like so:
```js
var pkg = {
  name: 'HACK',
  description: 'HackAday Cortext Kit',
  repository: {
    type: 'git',
    url: 'https://github.com/MickMad/HACK'
  }
}

var html = marky(yourMarkdown, {package: pkg})
```

Testing it out here locally just now, with the relative image paths, marky rendered:
```html
<p><img src="https://raw.githubusercontent.com/MickMad/HACK/master/graphics/hack01.png"></p>
<p><img src="https://raw.githubusercontent.com/MickMad/HACK/master/graphics/hack02.png"></p>
```

... which seems to be exactly what you're looking for. Does that work for you? If so, I think we can close this. Thanks again, and cheers! 👋

**Comment by kasbah** [comment]
Yes, sorry, I read about the `package` right after I filed this.

**Comment by kasbah** [comment]
After configuring the package repo and re-reading your instructions @revin, I think there is actually a valid use-case here. 

The linked README renders fine on GitHub but not with marky-markdown, even with the GitHub repo configured, because the img links, as they are, will only work on GitHub (I guess it's some server trickery). If you do "view image" on the GitHub README images in Firefox it will go to `raw.githubusercontent.com` and if you put up img links in html with a src as they appear anywhere but on GitHUb they won't work. They need to be changed to `githubusercontent.com` to work. 

I'll leave it in your court as to whether it's a use-case you want to support so I am re-opening the ticket.

**Comment by revin** [comment]
OK, great, thanks for the detail! We'll look into it. :+1:

**Comment by revin** [comment]
@kasbah hmm, after looking at this some more, I'm not sure this is a rendering issue. GitHub is doing the rendering as I posted above. I verified it with `curl`:

```
$ curl -s https://github.com/MickMad/HACK/ | grep hack01.png
<p><a href="https://github.com/MickMad/HACK/blob/master/graphics/hack01.png" target="_blank"><img src="https://github.com/MickMad/HACK/raw/master/graphics/hack01.png" alt="HACK top" style="max-width:100%;"></a></p>
```
So the URL is the same as the one in the markdown file, with no translation. It's just that when you request a raw image they do a 302 redirect to `raw.githubusercontent.com`. Here are the first few lines of the response:
```
$ curl -i https://github.com/MickMad/HACK/raw/master/graphics/hack01.png
HTTP/1.1 302 Found
Server: GitHub.com
Date: Sat, 13 Aug 2016 17:54:54 GMT
Content-Type: text/html; charset=utf-8
Status: 302 Found
Cache-Control: no-cache
Vary: X-PJAX, Accept-Encoding
X-RateLimit-Limit: 100
X-RateLimit-Remaining: 100
Access-Control-Allow-Origin: https://render.githubusercontent.com
Location: https://raw.githubusercontent.com/MickMad/HACK/master/graphics/hack01.png
X-UA-Compatible: IE=Edge,chrome=1
```
`*output snipped*`

@ashleygwilliams please overrule me if you want, but it seems to me like this is outside marky's domain.

**Comment by kasbah** [comment]
Makes me wonder why it works in the README. Doesn't seem to work inside issues either: 

![doesn't work](https://github.com/MickMad/HACK/blob/master/graphics/hack01.png)
source: `![doesn't work](https://github.com/MickMad/HACK/blob/master/graphics/hack01.png)`

![works](https://raw.githubusercontent.com/MickMad/HACK/master/graphics/hack01.png)
source: `![works](https://raw.githubusercontent.com/MickMad/HACK/master/graphics/hack01.png)`

**Comment by revin** [comment]
I wonder if it has something to do with the fact that the response is returning `Content-Type: text/html`

<img width="564" alt="screen shot 2016-08-13 at 11 20 17 am" src="https://cloud.githubusercontent.com/assets/94850/17644860/01af2fca-6148-11e6-86ec-0e44fd3ea0f4.png">

**Comment by revin** [comment]
@kasbah are you waiting on us for any other information here, or can we close this one out? As far as I can tell, the markdown rendering side of this is the same as GitHub, so I don't think there's anything for us to do here.

**Comment by kasbah** [comment]
Like I said, I'll leave it in your court. Personally, it's more important to me that the rendered page looks like it does on GitHub (rather than just the markup being exactly the same).

**Comment by revin** [comment]
ok thanks. 👍  Our current goals are to match GitHub's rendering, so it's out of scope for now; this might be the kind of thing we look at implementing once we've gotten to the point of parity.
